### PR TITLE
Lint fixes and linter config tweaks

### PR DESCRIPTION
**Pull request by karyon**
Boring stuff, mostly. review per commit.

After this we can have a look at Codacy and decide whether the current state is enough to consider #1092 fixed.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -218,7 +219,7 @@ class-attribute-naming-style=any
 
 # Regular expression matching correct class attribute names. Overrides class-
 # attribute-naming-style
-class-attribute-rgx=[a-z_][a-z0-9_]{1,30}$
+#class-attribute-rgx=
```
Comment (janno42) [inline .pylintrc]: i don't understand why this commit changes the settings in contrast to the one before?

File: evap/evaluation/migrations/0010_fill_textanswer_state.py [hunk 2]
```diff
@@ -28,6 +30,7 @@ def backward_update_states(apps, schema_editor):
             answer.checked = False
         answer.save()
 
+
 class Migration(migrations.Migration):
```
Comment (janno42) [inline evap/evaluation/migrations/0010_fill_textanswer_state.py]: could you also make these empty lines at the beginning of a class consistent? i'd prefer having none there.

File: .pylintrc [hunk 3]
```diff
@@ -254,7 +254,6 @@ good-names=i,
            _,
            __,
            e,
-           app_name,
```
Comment (janno42) [inline .pylintrc]: why was this introduced a few commits before and is removed here?

File: evap/evaluation/models.py [hunk 4]
```diff
@@ -583,7 +583,7 @@ def update_courses(cls):
 
 
 @receiver(post_transition, sender=Course)
-def log_state_transition(sender, instance, name, source, target, **kwargs):
+def log_state_transition(instance, name, source, target, **_kwargs):
```
Comment (janno42) [inline evap/evaluation/models.py]: using `**_kwargs` makes copying methods from other places in the code or from the documentation a little bit more work. i think i'd prefer listing `kwargs` in the `dummy-variables-rgx` and keeping the name the same everywhere regardless whether it's actually used in the method or not.

File: .pylintrc [hunk 5]
```diff
@@ -135,7 +135,6 @@ disable=print-statement,
         exception-escape,
         comprehension-escape,
         line-too-long,               # we don't enforce some fixed line length but rather complain about long lines in reviews.
-        cyclic-import,               # landscape re-reports them every time, https://github.com/PyCQA/pylint/issues/59
```
Comment (janno42) [inline .pylintrc]: this is another thing i don't really understand. we're not using landscape anymore, you introduced this a few commits before and remove it here.
so do we want to check for cyclic imports? could you then not add it in the other commit or did i miss something?

File: .pylintrc [hunk 6]
```diff
@@ -278,7 +278,8 @@ module-naming-style=snake_case
 
 # Regular expression matching correct module names. Overrides module-naming-
 # style
-#module-rgx=
+# allowing exactly four digits at the beginning for migrations
+module-rgx=([0-9]{4})?([a-z_][a-z0-9_]*)$
```
Comment (janno42) [inline .pylintrc]: sorry that i'm bringing this up again: couldn't this just be put in the commit "Apply our changes to .pylintrc"? i mean there are no other changes related to this that would make a separate commit necessary.

File: evap/evaluation/migrations/0055_reviewer_group.py [hunk 7]
```diff
@@ -2,11 +2,11 @@
 from django.db import migrations
 
 
-def add_group(apps, schema_editor):
+def add_group(_apps, _schema_editor):
```
Comment (janno42) [inline evap/evaluation/migrations/0055_reviewer_group.py]: as this is not only about `schema_editor` could you please rename the commit to `Mark unused arguments in migrations`?

**Comment by karyon** [comment]
the `new default .pylintrc` and `apply our changes` were supposed to just update the `.pylintrc` with the new defaults without any further changes, and in the following commits additional changes come in. i can make the second commit `Apply a completely new set of rules` instead of `apply [previous] changes`, i don't really care.

File: evap/evaluation/models.py (hunk 4 above)
Comment (karyon) [reply evap/evaluation/models.py]: by "a little bit more work" you mean adding an underscore? :)

i think there should be good reasons for ignoring parameters, and i don't see why that shouldn't apply to keyword arguments. additionally, in our codebase it's really just the receivers that are not using kwargs.

File: .pylintrc [hunk 8]
```diff
@@ -127,13 +127,23 @@ disable=print-statement,
...
         comprehension-escape,
+        line-too-long,               # we don't enforce some fixed line length but rather complain about long lines in reviews.
+        cyclic-import,               # landscape re-reports them every time, https://github.com/PyCQA/pylint/issues/59
+        too-many-public-methods,     # reported for some models, that won't change
+        too-few-public-methods,      # noisy, e.g. exception classes, mixins etc
+        no-member, no-name-in-module, # false positives, deals badly with django classes
+        pointless-string-statement,  # used for documentation in some test files
+        logging-format-interpolation, # decided against the proposed change in favor of .format()
+        too-many-ancestors,          # noisy, reported for all test classes
+        bad-continuation,            # Over a hundred reports, we seem to indent differently, but at least consistently.
```
[the remark this answers is not in the document]
Comment (karyon) [reply .pylintrc]: this commit applies the rules we had before. we can totally discuss this, and if we decide to follow whatever pylint wants, we can remove this exception and apply that style everywhere, but i really wouldn't do that in this PR. maybe you can open a new "style bikeshedding" issue? or put this into the static analysis one?

File: evap/evaluation/migrations/0010_fill_textanswer_state.py (hunk 2 above)
Comment (karyon) [reply evap/evaluation/migrations/0010_fill_textanswer_state.py]: i don't see us enforcing that during review, or would you do that? and i cannot find a lint that would enforce this. then i would rather not care about this altogether...

Comment (janno42) [reply evap/evaluation/migrations/0010_fill_textanswer_state.py]: i just thought since you're at it. sure, we definitely can't enforce that during review when there is no lint checking it because we will overlook it eventually. i mean in most cases i would notice it if someone adds it because it annoys me, but that's not guaranteed. okay then, look like we keep it.

File: evap/evaluation/models.py (hunk 4 above)
Comment (janno42) [reply evap/evaluation/models.py]: especially i mean thinking about anything in addition to the actual code. it is no problem at all for you and me, but a small one for less experienced contributors. the linter won't explain them what to do here and they might wonder why something they read in the docs doesn't fulfill the requirements. so it will always be a comment in the reviews explaining them what to do.
but as it makes the code more correct i'm on your side that we shouldn't handle them differently, so i'm okay with keeping the proposed solution.

**Comment by karyon** [comment]
sorted imports are already part of the lints. i have not, in fact, cleaned up everything, just some of the ones that were reported in large numbers, there are now about 300 lints reported instead of 600.
